File: src/CSP-AF/CSPCore/CSPRedwoodHQ/CSPCheckIfActionFailed.cs
using System;
using System.Linq;
using System.Collections.Generic;

using CloudSuitePortal;
using CloudSuitePortal.Services.RedwoodHQ;

namespace actions.CSP
{
    class CSPCheckIfActionFailed
    {
        public void run(Dictionary<string, object> Params)
        {
            var id = (string)Params["Id"];
            
            //var actionManager = new CSPActionManager();
           	//var action = actionManager.Action(a => a.Name == id);

            //if (action.CurrentState == CSPActionItem.State.NotCompleted)
            //{
            //    action.CurrentState = CSPActionItem.State.Failed;
            //    actionManager.SaveAction(action);
            //}
        }
    }
}

File: src/CSPAF/CSPCore/CSPRedwoodHQ/CSPCheckIfActionFailed.java
package CSPAF.CSPCore.CSPRedwoodHQ;

import java.util.*;

import com.infor.cloudsuite.portal.core.helpers.*;
import com.infor.cloudsuite.portal.core.services.*;

class CSPCheckIfActionFailed
{
    public void run(HashMap<String, Object> params) throws Exception
    {
        String id = params.get("Id").toString();
            
        CSPActionManager actionManager = new CSPActionManager();
        CSPActionItem action = actionManager.action(actionItem -> actionItem.name.equals(id));

        if (action == null) { throw new NullPointerException("The action could not be found!"); }
        
        if (action.currentState == CSPActionItem.State.NotCompleted)
        {
        	action.currentState = CSPActionItem.State.Failed;
        	actionManager.saveAction(action);
        }
    }
}